src/clp_ffi_js/ir/StreamReader.cpp

```diff
@@ -0,0 +1,221 @@
+#include "StreamReader.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <iterator>
+#include <memory>
+#include <span>
+#include <string>
+#include <string_view>
+#include <system_error>
+#include <utility>
+#include <vector>
+
+#include <clp/ErrorCode.hpp>
+#include <clp/ffi/encoding_methods.hpp>
+#include <clp/ffi/ir_stream/decoding_methods.hpp>
+#include <clp/ir/LogEvent.hpp>
+#include <clp/ir/LogEventDeserializer.hpp>
+#include <clp/ir/types.hpp>
+#include <clp/streaming_compression/zstd/Decompressor.hpp>
+#include <clp/TraceableException.hpp>
+#include <emscripten/em_asm.h>
+#include <emscripten/val.h>
+#include <spdlog/spdlog.h>
+
+#include <clp_ffi_js/ClpJsException.hpp>
+#include <clp_ffi_js/constants.hpp>
+
+using namespace std::literals::string_literals;
+
+namespace clp_ffi_js::ir {
+auto StreamReader::create(emscripten::val const& data_array) -> StreamReader {
+    auto const length{data_array["length"].as<size_t>()};
+    SPDLOG_INFO("StreamReader::StreamReader() got buffer of length={}", length);
+
+    // Copy array from JavaScript to C++
+    std::vector<char> data_buffer(length);
```

Comment:
I think we can use `clp::Array` to replace the usage of vector here from this PR: https://github.com/y-scope/clp/pull/513

Reply:
Sure, updated in 6527018cb2f07c1d2cfe0a06cc908de3e72b11fb . Could you take a look and see if the usages are correct?

Reply:
This was discussed offline with @LinZhihao-723 and the changes "should be no problem".